lib/board.rb

```diff
@@ -1,113 +1,101 @@
 class Board
-  attr_accessor :grid 
 
   def initialize(symbols = Array.new(9))
     @grid = symbols
   end
 
   def is_empty
-    !(grid.include?(:X) or grid.include?(:O))
+    !(grid.include?(PlayerSymbols::X) or grid.include?(PlayerSymbols::O))
   end
 
-  def update(index, symbol)
-    grid_before_index = copy_portion_of_grid(index)
-    grid_after_index = copy_portion_of_grid(grid.size - (index + 1))  
+  def make_move(index, symbol)
+    grid_before_index = copy_portion_of_grid(0, index)
+    grid_after_index = copy_portion_of_grid(index + 1, grid.size)   
 
-    set_grid Array.new(grid_before_index + [symbol] + grid_after_index)
+    Board.new(Array.new(grid_before_index + [symbol] + grid_after_index))
   end
 
   def has_free_spaces 
     grid.include?(nil)
   end
 
-  def has_winning_combination
-    all_winning_combinations = all_winning_combinations_for_3x3_grid
+  def get_symbol_at(position)
+    grid.at(position)
+  end
 
-    has_win_for_x = check_for_winning_row_of(:X, all_winning_combinations) 
-    has_win_for_o = check_for_winning_row_of(:O, all_winning_combinations)
+  def has_winning_combination
```

Comment:
Feels like there's some unnecessary delegation here. `has_winning_combination` calls `has_winning_combination_within` which calls `has_row_of_matching_symbols` which calls `find_winning_row_from` which actually has the behaviour we're interested in.